sa/type-converter.go

```diff
@@ -38,6 +38,16 @@ func (tc BoulderTypeConverter) ToDb(val interface{}) (interface{}, error) {
 		return string(t), nil
 	case core.OCSPStatus:
 		return string(t), nil
+	case *core.AcmeURL:
```

Comment:
Repetitive work can be saved with function calls. Depending so heavily on gorp to get things right in a very confusing bit of code is a problem. I already do not understand the code very well and cannot guarantee it will behave correctly. JSHA and I had to do some debugging in reflect to see why an issue with \*JSONBuffer occurred and it was very befuddling and ate up too much time.

This will also lead to things like we've had to fix before: folks not setting the schema's correctly or assuming certain things are true that are not (like the ordering of JWK's marshalling). This code is a small.